packages/paypal-commerce-integration/src/PaypalCommerceRatePayPaymentMethod.tsx

```diff
@@ -83,8 +88,31 @@ const PaypalCommerceRatePayPaymentMethod: FunctionComponent<any> = ({
                     legalTextContainer: 'legal-text-container',
                     getFieldsValues: () => fieldsValues.current,
                     onPaymentSubmission: (isSubmitting: boolean) => setIsPaymentSubmitting(isSubmitting),
-                    onError: (error: Error) => {
+                    onError: (error: SpecificError) => {
                         paymentForm.disableSubmit(method, true);
+                        const ratepaySpecificError = error?.errors?.filter(e => e.provider_error);
+
+                        if (ratepaySpecificError) {
+                            let translationCode;
+                            switch (ratepaySpecificError[0].provider_error?.code) {
+                                case PAYMENT_SOURCE_DECLINED_BY_PROCESSOR:
+                                    translationCode = 'paymentSourceDeclinedByProcessor';
+                                    break;
+                                case PAYMENT_SOURCE_INFO_CANNOT_BE_VERIFIED:
+                                    translationCode = 'paymentSourceInfoCannotBeVerified';
+                                    break;
+                            }
+
+                            const ratepayError = new CustomError({
+                                data: {
+                                    shouldBeTranslatedAsHtml: true,
+                                    translationKey: `payment.ratepay.errors.${translationCode}`
+                                },
+                            });
```

Comment:
What should be an error when `ratepaySpecificError[0].provider_error?.code` will not contain any code that you described in a switch case?

It will be something like this: `payment.ratepay.errors.`. It might be better to just pass an error from props as `ratepayError` if it is not covered in a `switch`, cause we don't know if there any changes on PP side

Reply:
fixed